### ISSUE TITLE
Autocompleter not sharing the same client pool

### ISSUE THREAD

**Issue by filipecosta90** [RediSearch/redisearch-go#35]
@gkorland @MeirShpilraien we should refactor this and remove:
```
// NewAutocompleter creates a new Autocompleter with the given host and key name
func NewAutocompleter(addr, name string) *Autocompleter
```
Autocompleter should only have:
```
// Autocompleter implements a redisearch auto-completer API
type Autocompleter struct {
	name string
}
```
instead of the current:
```
// Autocompleter implements a redisearch auto-completer API
type Autocompleter struct {
	pool *redis.Pool
	name string
}
```

**Comment by filipecosta90** [comment]
this issue is fixed by #36 with the method 
```func NewAutocompleterFromPool(pool *redis.Pool, name string) *Autocompleter```

**Comment by filipecosta90** [comment]
solved by #36
